include/deal.II/matrix_free/mapping_info.h

```diff
@@ -37,71 +37,260 @@ namespace internal
   namespace MatrixFreeFunctions
   {
     /**
-     * The class that stores all geometry-dependent data related with cell
-     * interiors for use in the matrix-free class.
+     * An enum to identify various types of cells. The most general type is
+     * what we typically compute in the FEValues context but for many
+     * geometries we can save significant storage.
+     */
+    enum CellType : unsigned char
+    {
+      /**
+       * The cell is Cartesian.
+       */
+      cartesian  = 0,
+      /**
+       * The cell may be described with an affine mapping.
+       */
+      affine     = 1,
+      /**
+       * The current face is flat, i.e., the normal factor on a face is the
+       * same on all quadrature points.
+       */
+      flat_faces = 2,
```

Comment:
This is a good comment and we should discuss this in detail: My plan is to use the same information field for both cells and faces. In that sense, the name `CellType` is misleading as it would rather be the type of the face. However, I want to use the same base class `FEEvaluationBase` for both cells and faces, distinguishing them only by a template argument `is_face` in the base class, where it is much simpler to have the same type - it is a simple integer after all. The end result are the two classes `FEEvaluation` and `FEFaceEvaluation` once derived but they share the mechanism of some data that gets re-used. 

The two options I see is to either rename this into `GeometryType` (rather than `CellType`) - it is inside an `internal` namespace, so we could do it - or to live with the fact that we have a misnomer. What do you think?

Reply:
> The two options I see is to either rename this into GeometryType (rather than CellType) - it is inside an internal namespace, so we could do it - or to live with the fact that we have a misnomer. What do you think?

My 2¢: Since this has been picked up early enough, IMO its worth the pain to fix the misnomer now rather than cause some confusion in the future and regret not having made the change. I'd vote for `GeometryType`.

Reply:
I also think that changing to `GeometryType` makes sense especially since we are not breaking anyone's code.